tests/spark/autologging/ml/test_pyspark_ml_autologging.py

```diff
@@ -723,6 +723,44 @@ def test_gen_estimator_metadata(spark_session):  # pylint: disable=unused-argume
     )
 
 
+@pytest.mark.parametrize("log_datasets", [True, False])
+def test_basic_post_training_datasets_autologging(dataset_iris_binomial, log_datasets):
+    mlflow.pyspark.ml.autolog(log_datasets=log_datasets)
+    estimator = LogisticRegression(maxIter=1, family="binomial", regParam=5.0, fitIntercept=False)
+
+    with mlflow.start_run() as run:
+        model = estimator.fit(dataset_iris_binomial)  # pylint: disable=unused-variable
+
+    run_id = run.info.run_id
+    client = MlflowClient()
+    dataset_inputs = client.get_run(run_id).inputs.dataset_inputs
+    if log_datasets:
+        assert len(dataset_inputs) == 1
+        assert dataset_inputs[0].dataset.source_type == "code"
+    else:
+        assert len(dataset_inputs) == 0
+
+
+def test_post_training_datasets_with_evaluate_autologging(dataset_iris_binomial):
+    mlflow.pyspark.ml.autolog(log_datasets=True, log_post_training_metrics=True)
+    estimator = LogisticRegression(maxIter=1, family="binomial", regParam=5.0, fitIntercept=False)
+    eval_dataset = dataset_iris_binomial.sample(fraction=0.3, seed=1)
+
+    with mlflow.start_run() as run:
+        model = estimator.fit(dataset_iris_binomial)
+        mce = MulticlassClassificationEvaluator(metricName="logLoss")
+        pred_result = model.transform(eval_dataset)
+        logloss = mce.evaluate(pred_result)  # pylint: disable=unused-variable
+
+    run_id = run.info.run_id
+    client = MlflowClient()
+    dataset_inputs = client.get_run(run_id).inputs.dataset_inputs
+    assert len(dataset_inputs) == 2
+    assert dataset_inputs[0].tags[0].value == "train"
+    assert dataset_inputs[1].dataset.name == "eval_dataset"
```

Comment:
Looks like we're using the variable name as the dataset name for pyspark evaluation autologging, but we aren't doing this anywhere else. I think we should just call it `dataset` for consistency with scikit-learn, etc.